### ISSUE TITLE
Workers should adjust BLAS/LAPACK/Numpy threads

### ISSUE THREAD

**Issue by ravwojdyla** [dask/dask-cloudprovider#230]
Should workers by default adjust BLAS/LAPACK/Numpy threads as per [dask-best-practices](https://docs.dask.org/en/latest/array-best-practices.html#avoid-oversubscribing-threads)? Since we have access to the number of threads/workers and cores, it should be possible adjust those accordingly. wdyt?

**Comment by jacobtomlinson** [comment]
Absolutely. There is work happening already in dask/distributed#4377 to address this.

**Comment by ravwojdyla** [comment]
@jacobtomlinson please correct me if I'm wrong, does dask/distributed#4377 really address this issue? Looking briefly into that, it seems like it's more about the number of Dask processes and thread but not the BLAS/LAPACK/Numpy threads?

**Comment by jacobtomlinson** [comment]
Ah right I see, I misunderstood the request.

Most cluster managers support an `env_vars` configuration option, so folks can set things however they want.

**Comment by ravwojdyla** [comment]
@jacobtomlinson thanks for prompt response. I understand, but do you think `dask-cloudprovider` could set sensible defaults given that it has information about number of Dask workers/threads and VM types (#CPUs)? wdyt?

**Comment by jacobtomlinson** [comment]
That's is definitely a possibility. I'm not sure whether we do this in any other cluster managers. Perhaps someone else from @dask/maintenance might know?

**Comment by martindurant** [comment]
Sounds like another point where spec-cluster might be able to give a consolidated view over different deployment types - like how many CPU cores are available to each worker, possibly different than the number of threads requested. I suspect, though, that in most cases setting all the threading options to 1 is the right thing to do.
